### PR TITLE
Auto Form

### LINKED ISSUE #337: [Feature]: Auto Form Generation from Zod/Yup schemas
### Describe the feature

https://github.com/vantezzen/auto-form

Porting this to VueJS might be great idea.

### Additional information

- [ ] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

### PR DESCRIPTION
**Pull request by zernonia**

### CODE REVIEW

**Comment by zernonia** [comment]
The feature were fully implemented. Left documentation. @sadeghbarati  do help look through the code and test out some cases ya 😁

**Comment by sadeghbarati** [comment]
zod `default()` is working for most of the fields but `Select` component is not selecting the default value in zod schema

Also, there are some unused imports which you can remove with `lint:fix` script before you want to merge this branch 💯

File: apps/www/src/lib/registry/default/ui/auto-form/dependencies.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+      switch (dep.type) {
+        case DependencyType.DISABLES:
+          if (conditionMet)
+            isDisabled.value = true
+
+          break
+        case DependencyType.REQUIRES:
+          if (conditionMet)
+            isRequired.value = true
```
Comment (sadeghbarati) [inline apps/www/src/lib/registry/default/ui/auto-form/dependencies.ts:69]: I saw a video from `vee-validate` author, about `useField` meta object which contains `required` value 
for handling `*` in form fields

https://x.com/logaretm/status/1728433759358210143?s=33

We can let `vee-validate` handle the `required` part, adding and resetting required value

**Comment by sadeghbarati** [comment]
I see Vue warn in my Devtools 

image

Do you think we can prevent this error like this? [`registry/default/ui/form/useFormField.ts`](https://github.com/radix-vue/shadcn-vue/blob/dev/apps/www/src/lib/registry/default/ui/form/useFormField.ts)

**Comment by sadeghbarati** [comment]
This feature is so nice 🔥optional custom wrapper/style/ui for `AutoFormField`s

image

**Comment by sadeghbarati** [comment]
This feature is so nice 🔥optional custom wrapper/style/ui for `AutoFormField`s

image

File: apps/www/src/lib/registry/default/ui/auto-form/dependencies.ts (hunk 1 above)
Comment (zernonia) [reply apps/www/src/lib/registry/default/ui/auto-form/dependencies.ts:69]: I tried using this. But somehow it is showing a wrong `required` value. Maybe I can re-check 😁

Comment (zernonia) [reply apps/www/src/lib/registry/default/ui/auto-form/dependencies.ts:69]: Tested again. Seems like the `required` is not really reliable. 

In cases of `enum` and other it doesn't reflect correctly. So will keep the original required implementation

image

**Pull request merged**